src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -80,22 +81,26 @@ object TypeCast {
     } else {
       datum
     }
-
-    dataType match {
-      case NullType => castTo(value, StringType, options)
-      case LongType => signSafeToLong(value, options)
-      case DoubleType => signSafeToDouble(value, options)
-      case BooleanType => castTo(value, BooleanType, options)
-      case StringType => castTo(value, StringType, options)
-      case DateType => castTo(value, DateType, options)
-      case TimestampType => castTo(value, TimestampType, options)
-      case FloatType => signSafeToFloat(value, options)
-      case ByteType => castTo(value, ByteType, options)
-      case ShortType => castTo(value, ShortType, options)
-      case IntegerType => signSafeToInt(value, options)
-      case dt: DecimalType => castTo(value, dt, options)
-      case _ =>
-        sys.error(s"Failed to parse a value for data type $dataType.")
+    try {
+      dataType match {
+        case NullType => castTo(value, StringType, options)
+        case LongType => signSafeToLong(value, options)
+        case DoubleType => signSafeToDouble(value, options)
+        case BooleanType => castTo(value, BooleanType, options)
+        case StringType => castTo(value, StringType, options)
+        case DateType => castTo(value, DateType, options)
+        case TimestampType => castTo(value, TimestampType, options)
+        case FloatType => signSafeToFloat(value, options)
+        case ByteType => castTo(value, ByteType, options)
+        case ShortType => castTo(value, ShortType, options)
+        case IntegerType => signSafeToInt(value, options)
+        case dt: DecimalType => castTo(value, dt, options)
+        case _ =>
+          sys.error(s"Failed to parse a value for data type $dataType.")
+      }
+    }
+    catch {
+      case NonFatal(_) if options.permissive => null
```

Comment:
Hmmmmm.. IIRC Spark's JSON doesn't also allow the partial results. Also, wouldn't this fail to fill up the corrupt column?